Enhance: make compact input line feature act on individual profiles and be saved for each profile

SlySven opened the pull request:
Makes the compact input line a toggle (so now it also has constant text) and act on ~~all profiles~~ the currently active profile.  The previous code only considered the currently active profile which is contradicted by the fact that the state - which was maintained between sessions - was stored centrally as an application setting rather than a per profile one.

This was split out of a larger PR that originally included it: #3844 .

*Edit: It has been revised from acting on all profile simultaneously and storing that setting with the application's setting into acting individually and storing the state in the profile save (XML) file.  The state of the checkable item in the menu is now updated to reflect the state of the currently active profile when multi-playing and should change itself as the user switches between profiles. ~~Note that Mudlet's idea of which the currently active profile is, is a bit sloppy at the moment but it will should be fixed when #3844 is also merged in.~~*

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Enhance: make compact input line act on all profiles

Makes the compact input line a toggle (so now it also has constant text)
and act on all profiles.  The previous code only considered the currently
active profile which is contradicted by the fact that the state - which was
and is maintained between sessions - is stored centrally as an application
setting rather than a per profile one.

This was split out of a larger PR that originally included it:
https://github.com/Mudlet/Mudlet/pull/3844 .

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap commented:
There was some discussion back in the original PR already, and I feel it needs more discussion before it's settled, so thanks for seperating this out for now.

While it may actually have been a global setting (I did not check before, but gladly believe it), it never felt as one to the players. Why so?

When you open 2-3 profiles for playing, and only at some time want to toggle the input line, then it only affects this very profile. When you look at other profiles, they stay unchanged.

The only time you would notice the global change that happened under the surface, is if you opened a 4th profile during this time.

My usual workflow is like this: I have input line set to compact style always. Only if I need to search something, I toggle the extra buttons visible, use search, then hide them away again. If I use other profiles in between, I would not want or need the buttons there.

While I understand the additional effort required to make this an actual profile-specific setting, I think the effort is worth the result of not confusing players established expectations at this point.

SlySven committed:
Revise: make compact input line be a per profile option

Because of the current poor tracking of `(Host*) mudlet::mCurrentActiveHost`
the use of the short-cut or menu item to toggle the state of the input line
does not fully track which profile is active yet. That will improve when

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_makeCompactInputLineActOnAllProfiles

Conflicts resolved in:
* src/mudlet.cpp

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
~~I have resolved the conflict but have detected an issue in not updating the menu item checked status when clicking from one tab to another (which bypasses the `mudlet::slot_tab_changed(...))` because the tab-bar is NOT the recipient of the click...~~ Fixed.

SlySven committed:
BugFix: clicking on a different profile needs to update compact input line

Although clicking on the tab bar changed the currently active host and that
would update the compact input line menu item state to match the newly made
active profile the same thing was not happening if the profile was made
active by being clicked on in other places when in multi-view.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_makeCompactInputLineActOnAllProfiles

Needed to import fix for CI build failures on Travis MacOs

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap commented:
/refresh links

keneanung commented:
/refresh links

Kebap commented:
/refresh links

ah well, time to access https://make.mudlet.org/snapshots/ myself and search for 3866 there :)

Kebap commented on src/ui/main_window.ui:
```diff
@@ -306,8 +306,14 @@
+   </property>
    <property name="toolTip">
-    <string>&lt;p&gt;Hide or show the search area and the bottom buttons to the right of the input area on the input line.&lt;/p&gt;</string>
+    <string>&lt;p&gt;Hide or show the search area and the bottom buttons to the right of the input area on the input line for all profiles.&lt;/p&gt;</string>
```
Not "for all profiles" anymore!

SlySven replied:
Fixed by direct edit here...

Kebap commented:
This PR does what it promises.

SlySven commented on src/ui/main_window.ui:
```diff
@@ -306,8 +306,14 @@
+   </property>
    <property name="toolTip">
-    <string>&lt;p&gt;Hide or show the search area and the bottom buttons to the right of the input area on the input line.&lt;/p&gt;</string>
+    <string>&lt;p&gt;Hide or show the search area and the bottom buttons to the right of the input area on the input line for all profiles.&lt;/p&gt;</string>
```
```suggestion
    <string>&lt;p&gt;Hide or show the search area and the bottom buttons to the right of the input area on the input line for the current profile.&lt;/p&gt;</string>
```

SlySven committed:
Update src/ui/main_window.ui

Revise: fixup a tool-tip to match the situation now

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
